### PR TITLE
✨ JSON MultiType Support

### LINKED ISSUE #135: Support JSON In MultiType
What it says on the tin. Just creating a separate issue since it's better tracked away from the XML work I started it in. But is basically required for that to be considered complete.

### PR DESCRIPTION
**Pull request by effs**
JSON support for MultiType (incl. w/ XML), plus probably refactors around that.

- Added test helper `mergeOrFailOnConflict(Map<K, V>, Map<K, V>)` which does just that.
- Removed the special-casing for AMM JSONs (but left the test ofc), we just don't look there

Also:

- `wrapInstall` now checks for filecount mismatch and invokes the Fallback installer if there is a mismatch. Would need to remove any logic that currently rejects on a mismatch somewhere higher up and let it just get picked up here instead. And should probably add a test for this but currently there's nothing that triggers this case. (And with the existing tests it seems to work correctly for the cases where it's not hit.)

Closes #135

### CODE REVIEW

**Comment by E1337Kat** [comment]
Just a note to change the base to `main` before actually doing the merge.

File: test/unit/utils.helper.ts [hunk 1]
```diff
@@ -56,6 +56,19 @@ export interface ExamplesForType {
...
 }
 
+const mapHasAnySameKeys = <K, V>(map1: Map<K, V>, map2: Map<K, V>): boolean =>
+  [...map1].some(([k, _]) => map2.get(k));
+
+// It's tests, it's ok to just raise. Don't do this in real code, kids
+export const mergeOrFailOnConflict = <K, V>(...maps: Map<K, V>[]): Map<K, V> =>
+  maps.reduce((mergedMap, map) => {
+    if (mapHasAnySameKeys(map, mergedMap)) {
+      // :goose-loose:
```
Comment (E1337Kat) [inline test/unit/utils.helper.ts:66]: :duck: :duck:

**Review by E1337Kat — APPROVED** [review]
lg2m so far

**Comment by effs** [comment]
I still need to figure out how to test that we get two prompts on XML + JSON success, and which prompt we get on XML+JSON failure.

**Review by E1337Kat — APPROVED** [review]
looks good.

**Pull request merged**